controllers/flowlogspipeline/flp_objects.go

```diff
@@ -252,13 +265,83 @@ func (b *builder) addTransformStages(lastStage *config.PipelineBuilderStage) {
 	}
 
 	// prometheus stage (encode) configuration
+	// the aggregate and prometheus rules will be incorporated from the metrics definitions file
 	agg := enrichedStage.Aggregate("aggregate", []api.AggregateDefinition{})
 	agg.EncodePrometheus("prometheus", api.PromEncode{
 		Port:   int(b.desired.PrometheusPort),
-		Prefix: "flp_",
+		Prefix: "netobserv_",
 	})
 }
 
+var generateStages = []string{"extract_aggregate", "encode_prom"}
+
+func (b *builder) mergeMetricsConfiguration(pipeline config.PipelineBuilderStage) {
+	// set confgen.Opt.SrcFolder, etc
+	confgen.Opt.SrcFolder = b.desired.MetricsDefinitionsDir
+	confgen.Opt.DestConfFile = "/dev/null"
+	confgen.Opt.SkipWithTags = b.desired.IgnoreMetrics
+	confgen.Opt.GenerateStages = generateStages
+
+	if confgen.Opt.SrcFolder == "" {
+		return
+	}
+	// run the confgenerator to produce the proper flp configuration for metrics from metrics_definitions
+	cg, err := confgen.NewConfGen()
+	if err != nil {
+		log.Fatalf("failed to initialize NewConfGen %s", err)
+		return
+	}
+	err = cg.Run()
+	if err != nil {
+		log.Fatalf("failed to initialize NewConfGen %s", err)
```

Comment:
I guess it's "fail to run" here
```suggestion
		log.Fatalf("failed to run NewConfGen %s", err)
```

Reply:
fixed